src/app/drive/components/ShareInviteDialog/ShareInviteDialog.tsx

```diff
@@ -0,0 +1,231 @@
+import isValidEmail from '@internxt/lib/dist/src/auth/isValidEmail';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import Button from 'app/shared/components/Button/Button';
+import Avatar from 'app/shared/components/Avatar';
+import BaseCheckbox from 'app/shared/components/forms/BaseCheckbox/BaseCheckbox';
+import { useTranslationContext } from 'app/i18n/provider/TranslationProvider';
+import './ShareInviteDialog.scss';
+
+interface ShareInviteDialog {
+  onInviteUser: () => void;
+}
+
+interface UsersToInvite {
+  email: string;
+  userRole: string;
+}
+
+const ShareInviteDialog = (props: ShareInviteDialog): JSX.Element => {
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    watch,
+    resetField,
+  } = useForm<IFormValues>({ mode: 'onChange' });
+  const { translate } = useTranslationContext();
+  const emailRegex =
+    /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
+  const email = watch('email');
+  const [userRole, setUserRole] = useState<string>('editor');
+  const [usersToInvite, setUsersToInvite] = useState<Array<UsersToInvite>>([]);
+  const [notifyUser, setNotifyUser] = useState<boolean>(false);
+  const [messageText, setMessageText] = useState<string>('');
+  const [isInviteButtonDisabled, setIsInviteButtonDisabled] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (email) {
+      isValidEmail(email) || usersToInvite.length > 0
+        ? setIsInviteButtonDisabled(false)
+        : setIsInviteButtonDisabled(true);
+    }
+  }, [email]);
+
+  const onAddInviteUser: SubmitHandler<IFormValues> = (formData, event) => {
+    event?.preventDefault();
+    const userInvitedEmail = formData.email;
+    const userInvitedRole = formData.userRole;
+    const userInvited = { email: userInvitedEmail, userRole: userInvitedRole };
+    const isDuplicated = usersToInvite.find((user) => user.email === userInvited.email);
+
+    if (!isDuplicated) {
+      const unique: Array<UsersToInvite> = [...usersToInvite];
+      unique.push(userInvited);
+      setUsersToInvite(unique);
+    }
+    resetField('email');
+  };
+
+  const onEditRole = () => {
+    //   Edit user role
+  };
+
+  const onInvite = () => {
+    // Ivite added users
+  };
+
+  return (
+    <div>
+      <div className="flex h-16 w-full items-center justify-between space-x-5 border-b border-gray-10 px-5">
+        <h4
+          className="flex max-w-full items-center space-x-4 overflow-hidden overflow-ellipsis whitespace-nowrap text-xl font-medium"
+          title={translate('modals.shareModal.invite.title')}
+        >
+          <ArrowLeft onClick={props.onInviteUser} size={22} className="cursor-pointer" />
+          <span>{translate('modals.shareModal.invite.title')}</span>
+        </h4>
+      </div>
+
+      <div className="p-5">
+        <form className="m flex w-full" onSubmit={handleSubmit(onAddInviteUser)}>
+          <input
```

Comment:
1. Input used here should be:
`import Input from 'app/shared/components/Input';`

2. If the user types a comma "," it should have the same effect as pressing enter (add it to the "Users to invite" list and empty the input)

Reply:
Done